### PR TITLE
Zero-terminated C strings

### PR DESCRIPTION
**Pull request by tahina-pro**
`T field_name[:string]` 
`T field_name[:string-byte-size-at-most size]` 

where `T` is the element type, and `size` is the size in bytes (in the latter case, the input buffer will consume `size` bytes but is expected to start with a maybe shorter zero-terminated string.)

The zero terminator is automatically deduced from `T` (currently assuming that `T` is a numerical type) because I claim that it would be too heavy to mandate the user to manually specify the zero every time they want a string.

### CODE REVIEW

**Comment by nikswamy** [comment]
Thanks! THis is very useful. 

How about using the qualifier keywords "zero-terminated-string" and "zero-terminated-string-at-most-byte-size". It's long but very explicit

File: src/3d/Ast.fst [hunk 1]
```diff
@@ -286,12 +286,18 @@ type array_qualifier =
   | ArrayByteSizeAtMost //[:byte-size-at-most
   | ArrayByteSizeSingleElementArray //[:byte-size-single-element-array
 
+noeq
+type field_array_t =
+  | FieldArrayNormal
```
Comment (nikswamy) [inline src/3d/Ast.fst:291]: this case indicates that the field is not an array, right?

Comment (tahina-pro) [reply src/3d/Ast.fst:291]: yes, the field is neither a string nor otherwise an array

**Comment by tahina-pro** [comment]
> How about using the qualifier keywords "zero-terminated-string" and "zero-terminated-string-at-most-byte-size". It's long but very explicit

I'll do `zeroterm`. (I also proposed `cstring` but usually C strings are only byte strings.)

**Comment by nikswamy** [comment]
thanks! merging

**Pull request merged**